core/dive.c

```diff
@@ -1880,6 +1882,42 @@ static char *merge_text(const char *a, const char *b)
 	return res;
 }
 
+static char *merge_csv(const char *a, const char *b)
+{
+	char *res;
+	if (!a && !b)
+		return NULL;
+	if (!a || !*a)
+		return copy_string(b);
+	if (!b || !*b)
+		return strdup(a);
+	if (!strcmp(a, b))
+		return copy_string(a);
+	res = malloc(strlen(a) + strlen(b) + 32);
+	if (!res)
+		return (char *)a;
+	sprintf(res, "%s, %s", a, b);
+	return res;
+}
+
+static char *merge_notes(const char *a, const char *b)
+{
+	char *res;
+	if (!a && !b)
+		return NULL;
+	if (!a || !*a)
+		return copy_string(b);
+	if (!b || !*b)
+		return strdup(a);
+	if (!strcmp(a, b))
+		return copy_string(a);
+	res = malloc(strlen(a) + strlen(b) + 32);
+	if (!res)
+		return (char *)a;
+	sprintf(res, "%s<br>%s", a, b);
+	return res;
+}
```

Comment:
given that this is almost the same code twice, why don't you make a single merge function that takes a separator string as third argument and sprintfs that between a and b?

Reply:
^ +1

Reply:
and/or `merge_text`, `merge_notes`, `merge_csv` can pretty much use the same `static` helper function as they share the same code.